ddl/column.go

```diff
@@ -879,16 +879,30 @@ func replaceOldIndexes(tblInfo *model.TableInfo, changingIdxs []*model.IndexInfo
 	}
 }
 
-func updateNewIndexesCols(tblInfo *model.TableInfo, oldName, newName model.CIStr, newOffset int) {
-	for _, idx := range tblInfo.Indices {
+// updateNewIndexesCols updates the changing indexes column name&offset, and
+// filter out the indexes that can be removed.
+func updateNewIndexesCols(tblInfo *model.TableInfo, changingIdxs []*model.IndexInfo,
```

Comment:
This function name should be refined.  It's strange for `updateNewIndexesCols` to return a value.
For example, `updateXXXAndYYY`.

Reply:
OK. I have split it into two functions.